python/lsst/cell_coadds/_fits.py

```diff
@@ -238,7 +238,7 @@ def readAsMultipleCellCoadd(self) -> MultipleCellCoadd:
 
             # Build the quantities needed to construct a MultipleCellCoadd.
             common = CommonComponents(
-                units=CoaddUnits(1),  # TODO: read from FITS TUNIT1 (DM-40562)
+                units=getattr(CoaddUnits, header["TUNIT1"]),
```

Comment:
Making `CoaddUnits` a child class of `enum.StrEnum` instead of `enum.Enum` almost achieves this. The only exception is it modifies `nJy` to `njy`

```python
print(CoaddUnits('njy'))
<CoaddUnits.nJy: 'njy'>
```

Looking into it.

Reply:
https://github.com/python/cpython/blob/2938c3dec99390087490124c2ef50e1592671e72/Lib/enum.py#L1366-L1371

makes the lower case version of it as values. I could make this line be `CoaddUnits(header["TUNIT1"].lower())`. Not as nice as it could have been, but better than the current change.

Reply:
Oh, I see what you meant. Just using these strings explicitly instead of `enum.auto` is good.